### ISSUE TITLE
Next throws on cancelled renders when accessing dynamic route url on app first load

### ISSUE THREAD

**Issue by phamhieu** [vercel/next.js#18003]
# Bug report

## Describe the bug

After refactoring our frontend to support [Automatic Static Optimization](https://nextjs.org/docs/advanced-features/automatic-static-optimization), we received lot of **Cancel rendering route** errors on Sentry.

Whenever users access a **dynamic route** pages like `/org/[slug]/settings.js` on app first load, an error shows on browser console `Uncaught (in promise) Error: Cancel rendering route`. The error triggers only 1 time, after that everything works properly.

If users access static route pages like `/about` on app first load then it is totally fine. There is no error.

I found a related closed issues https://github.com/vercel/next.js/issues/16445

## To Reproduce

There's no error on dev mode. Only after deploying to production, the error starts to appear.

```
Error: Cancel rendering route
  at G(/_next/static/chunks/main-b2a1c3544d25cb81d913.js:1:8918)
  at fe/s<(/_next/static/chunks/main-b2a1c3544d25cb81d913.js:1:8858)
  at fe(/_next/static/chunks/main-b2a1c3544d25cb81d913.js:1:8828)
  at e/<(/_next/static/chunks/main-b2a1c3544d25cb81d913.js:1:6521)
  at s(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:24546)
  at c/a._invoke</<(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:24300)
  at m/</<(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:24905)
  at r(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:33915)
  at u(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:34127)
  at yXPU/t.exports/</<(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:34186)
  at yXPU/t.exports/<(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:34066)
  at ne(/_next/static/chunks/main-b2a1c3544d25cb81d913.js:1:6275)
  at subscription(/_next/static/chunks/main-b2a1c3544d25cb81d913.js:1:5906)
  at value(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:19901)
  at value(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:16933)
  at e/<(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:14019)
  at s(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:24546)
  at c/a._invoke</<(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:24300)
  at m/</<(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:24905)
  at r(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:33915)
  at u(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:34127)
  at promise callback*r(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:33986)
  at u(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:34127)
  at promise callback*r(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:33986)
  at u(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:34127)
  at promise callback*r(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:33986)
  at u(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:34127)
  at yXPU/t.exports/</<(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:34186)
  at yXPU/t.exports/<(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:34066)
  at elyg/C</<.value</<(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:14452)
  at value(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:11431)
  at value(/_next/static/chunks/main-b2a1c3544d25cb81d913.js:1:4346)
  at oa(/_next/static/chunks/framework.78ebc23da16188abcacc.js:1:115626)
  at hu(/_next/static/chunks/framework.78ebc23da16188abcacc.js:1:133555)
  at +wdc/t.unstable_runWithPriority(/_next/static/chunks/framework.78ebc23da16188abcacc.js:1:3950)
  at $l(/_next/static/chunks/framework.78ebc23da16188abcacc.js:1:77050)
  at pu(/_next/static/chunks/framework.78ebc23da16188abcacc.js:1:130026)
  at eu(/_next/static/chunks/framework.78ebc23da16188abcacc.js:1:126155)
  at Ya(/_next/static/chunks/framework.78ebc23da16188abcacc.js:1:122717)
  at Lu(/_next/static/chunks/framework.78ebc23da16188abcacc.js:1:144226)
  at $u/<(/_next/static/chunks/framework.78ebc23da16188abcacc.js:1:145473)
  at nu(/_next/static/chunks/framework.78ebc23da16188abcacc.js:1:126303)
  at $u(/_next/static/chunks/framework.78ebc23da16188abcacc.js:1:145458)
  at yl30/t.hydrate(/_next/static/chunks/framework.78ebc23da16188abcacc.js:1:148369)
  at fe(/_next/static/chunks/main-b2a1c3544d25cb81d913.js:1:10235)
  at e/<(/_next/static/chunks/main-b2a1c3544d25cb81d913.js:1:6521)
  at s(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:24546)
  at c/a._invoke</<(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:24300)
  at m/</<(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:24905)
  at r(/_next/static/chunks/fe2b3b2fbb427f1012c46de54fbc0d90dafb47e7.1b0e03788822db0a5092.js:1:33915)
```

## Screenshots

<img width="451" alt="Screenshot 2020-10-19 at 10 46 17 AM" src="https://user-images.githubusercontent.com/689843/96396024-b7e91700-11f8-11eb-9211-c2bb2623472e.png">

## System information

- OS: macOS
- Browser chrome, firefox
- Version of Next.js: ^9.5.4
- Version of Node.js: latest LTS Node.js version available on Vercel

**Comment by timneutkens** [comment]
Please provide a reproduction so that we can have a look. Based on the error it seems that an error was thrown during hydration in your application 🤔

**Comment by phamhieu** [comment]
Hi @timneutkens, I found the line of code that trigger the error in my app.
```
// To suppress Warning: useLayoutEffect does nothing on the server
React.useLayoutEffect = React.useEffect
```
As my app already supports Automatic Static Optimization, I don't need to worry about that warning anymore. Removing that line also makes `Error: Cancel rendering route` disappeared.

I close the issue now. Sorry for waiting your time.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
